orchagent/portsorch.cpp

```diff
@@ -5981,34 +6033,69 @@ void PortsOrch::generateQueueMap(map<string, FlexCounterQueueStates> queuesState
                 FlexCounterQueueStates flexCounterQueueState(maxQueueNumber);
                 queuesStateVector.insert(make_pair(it.second.m_alias, flexCounterQueueState));
             }
-            generateQueueMapPerPort(it.second, queuesStateVector.at(it.second.m_alias));
+            generateQueueMapPerPort(it.second, queuesStateVector.at(it.second.m_alias), false);
+            if (gMySwitchType == "voq")
+            {
+               generateQueueMapPerPort(it.second, queuesStateVector.at(it.second.m_alias), true);
+            }
+        }
+        if (it.second.m_type == Port::SYSTEM)
+        {
+            if (!queuesStateVector.count(it.second.m_alias))
+            {
+                auto maxQueueNumber = getNumberOfPortSupportedQueueCounters(it.second.m_alias);
```

Comment:
the `getNumberOfPortSupportedQueueCounters` is getting the `m_queue_ids.size()` for the port. 
For system port what will be the size of `m_queue_ids`. I dont see `m_queue_ids` getting updated for system port

Reply:
The size will be zero but since queuesStateVector is not used for system ports, we do not have any issue.

Reply:
yes we will see the voq counters in counters db. I have ported https://github.com/sonic-net/sonic-utilities/pull/1827 in our env. I will create a pr for that too. That way you can see the voq counters using queuestat

Reply:
Created https://github.com/sonic-net/sonic-utilities/pull/2591 for the queuestat enhancement.

Reply:
https://github.com/sonic-net/sonic-swss/files/10388072/voq.txt has the output with the sai version of my chassis and output from queuestat -V. This is using the image with voq changes and fabric counter changes removed.